utils/utils.c

```diff
@@ -886,6 +886,23 @@ char *absolute_dirname(const char *path, char *resolved_path)
 	return resolved_path;
 }
 
+char *uftrace_strerror_r(int errnum, char *buf, size_t buflen)
+{
+	long strerror_r_result = (long) strerror_r(errnum, buf, buflen);
+
+	if (strerror_r_result == 0)
+		/* XSI-compliant strerror_r succeed */
+		return buf;
+	else if (strerror_r_result > 0 && strerror_r_result < 4096) {
+		/* XSI-compliant strerror_r failed */
+		sprintf(buf, "error: %ld", strerror_r_result);
```

Comment:
Please use `snprintf()` rather than plain `sprintf()`.  Also I'd like to print the errnum here (but printing strerror_r_result too looks ok).